### PR TITLE
Refactor GET requests

### PR DESCRIPTION
**Pull request by lj894**
When user call GET with a valid product_id, make sure the API returns [] instead of 404 error if no matching records found.

### CODE REVIEW

File: service/service.py [hunk 1]
```diff
@@ -210,17 +229,80 @@ def get_related_products_with_type(product_id, type_id):
...
-    products = []
-    product_status = []
+    active_products, inactive_products = [], []
     for recommendation in recommendations:
-        products.append(recommendation.related_product_id)
-        product_status.append(recommendation.status)
-
-    result = {"ids": products, "status": product_status}
+        if recommendation.status == "True":
+            active_products.append(recommendation.related_product_id)
```
Comment (sushanthsamala) [inline service/service.py:235]: Unit tests did not touch this line

File: service/service.py [hunk 2]
```diff
@@ -210,17 +229,80 @@ def get_related_products_with_type(product_id, type_id):
...
+        else:
+            inactive_products.append(recommendation.related_product_id)
+    result = []
+    if(len(active_products)):
+        result.append(
```
Comment (sushanthsamala) [inline service/service.py:240]: Unit tests did not touch this line

File: service/service.py [hunk 3]
```diff
@@ -136,14 +134,32 @@ def get_related_products(product_id):
...
             type_3_active.append(p.related_product_id) if p.status else type_3_inactive.append(p.related_product_id)
 
-    relationships = [
-        {"relation_id": 1, "ids": type_1_active, "inactive_ids": type_1_inactive},
-        {"relation_id": 2, "ids": type_2_active, "inactive_ids": type_2_inactive},
-        {"relation_id": 3, "ids": type_3_active, "inactive_ids": type_3_inactive}
-    ]
+
+    if len(type_1_active) or len(type_1_inactive):
+        relationships.append(
```
Comment (sushanthsamala) [inline service/service.py:139]: Unit tests did not touch this line

**Review by sushanthsamala — CHANGES_REQUESTED** [review]

File: service/service.py (hunk 1 above)
Comment (lj894) [reply service/service.py:235]: Added test cases.

**Review by sushanthsamala — APPROVED** [review]

File: service/service.py (hunk 2 above)
Comment (lj894) [reply service/service.py:240]: Added test cases.

File: service/service.py (hunk 3 above)
Comment (lj894) [reply service/service.py:139]: Added test cases.

**Pull request merged**